### PR TITLE
[TLI][AArch64] Add extra SLEEF mappings and tests

### PR DESCRIPTION
**Pull request by mgabka**
This patch is adding more scalar to vector mappings to TLI for SLEEF vector library.
The added mappings are for the following functions:

 acosh, asinh, cbrt, copysign, cospi
 erf, erfc, expm1, fdim, fma, fmax, fmin
 hypot, ilogb, ldexp, log1p, nextafter, sinpi

### CODE REVIEW

File: llvm/test/Transforms/LoopVectorize/AArch64/veclib-function-calls.ll [hunk 1]
```diff
@@ -2349,6 +2645,79 @@ for.cond.cleanup:
   ret void
 }
 
+declare void @sincospi(double, ptr, ptr)
+declare void @sincospif(float, ptr, ptr)
+
+define void @test_sincospi(ptr noalias %a, ptr noalias %b, ptr noalias %c) {
```
Comment (paschalis-mpeis) [inline llvm/test/Transforms/LoopVectorize/AArch64/veclib-function-calls.ll:2651]: nit: use `sincospi_f64` so it uses the same naming convention with the remaining tests

File: llvm/test/Transforms/LoopVectorize/AArch64/veclib-function-calls.ll [hunk 2]
```diff
@@ -2349,6 +2645,79 @@ for.cond.cleanup:
...
+  call void @sincospi(double %num, ptr %gepb, ptr %gepc)
+  %indvars.iv.next = add nuw nsw i64 %indvars.iv, 1
+  %exitcond = icmp eq i64 %indvars.iv.next, 1000
+  br i1 %exitcond, label %for.cond.cleanup, label %for.body
+
+for.cond.cleanup:
+  ret void
+}
+
+define void @test_sincospif(ptr noalias %a, ptr noalias %b, ptr noalias %c) {
```
Comment (paschalis-mpeis) [inline llvm/test/Transforms/LoopVectorize/AArch64/veclib-function-calls.ll:2686]: nit: use `sincospi_f32` so it uses the same naming convention with the remaining tests

File: llvm/test/Transforms/LoopVectorize/AArch64/veclib-function-calls.ll (hunk 1 above)
Comment (paschalis-mpeis) [reply llvm/test/Transforms/LoopVectorize/AArch64/veclib-function-calls.ll:2651]: Maybe worth to mention in the commit message that missing tests of `sincospi` are also added by this patch.

**Review by paschalis-mpeis — APPROVED** [review]
LGTM.

File: llvm/test/Transforms/LoopVectorize/AArch64/veclib-function-calls.ll (hunk 1 above)
Comment (mgabka) [reply llvm/test/Transforms/LoopVectorize/AArch64/veclib-function-calls.ll]: sure, will mention when merging the patch

Comment (labrinea) [reply llvm/test/Transforms/LoopVectorize/AArch64/veclib-function-calls.ll]: The tests for sincospi were accidentally removed in #77609. Good we are reinstating them here.

**Comment by labrinea** [comment]
Do we not want tests in ```llvm/test/Transforms/Util/add-TLI-mappings.ll``` ?

File: llvm/test/Transforms/LoopVectorize/AArch64/veclib-function-calls.ll (hunk 1 above)
Comment (mgabka) [reply llvm/test/Transforms/LoopVectorize/AArch64/veclib-function-calls.ll]: > The tests for sincospi were accidentally removed in #77609. Good we are reinstating them here.

@labrinea my understanding was that those tests are checking the inject-with-tli pass in isolation what is good to check if the attributes are correctly added on few test cases, while the llvm/test/Transforms/LoopVectorize/AArch64/veclib-function-calls.ll tests are primarily testing LoopVectorizer, they also call the inject-tli-mappings pass, so we have a testing of a whole flow.

I am not fully convinced it is worth to have a 100% coverage in  llvm/test/Transforms/Util/add-TLI-mappings.ll, as that would also include all veclibs for all other targets, and intriduce quite a lot of duplicated testing.

I would rather not add extra tests there, unless we are expecting changes in the form of the **vector-function-abi-variant** or how it is emitted.
For example it had sense to addd tests for sincos as there was no tests there for functions taking linear pointers.

Comment (labrinea) [reply llvm/test/Transforms/LoopVectorize/AArch64/veclib-function-calls.ll]: Ok I see your point, we wanted to test a new type of variants (those taking linear pointers) just once. Sounds good to me.

**Review by labrinea — APPROVED** [review]

**Pull request merged**